app/src/main/java/com/example/caregiver/Task_Caregivee.java

```diff
@@ -0,0 +1,195 @@
+        JsonParser parser = new JsonParser();
+        JsonObject roomObject = (JsonObject) parser.parse(roomString);
+        List < String > rooms = roomObject.entrySet().stream().map(
+                i ->i.getKey()).collect(Collectors.toCollection(ArrayList::new));
+
+        // For each room, get their corresponding tasks
+        for (String roomStr: rooms) {
+            JsonObject singleRoom = roomObject.getAsJsonObject(roomStr);
+            JsonObject tasksPerRoom = singleRoom.getAsJsonObject("tasks");
+            if (tasksPerRoom != null) {
+                List < String > tasksIds = tasksPerRoom.entrySet().stream().map(
+                        i ->i.getKey()).collect(Collectors.toCollection(ArrayList::new));
+
+                // For each task, put them in the Task object.
+                for (String taskId: tasksIds) {
+                    Log.d("taskId", taskId);
+                    JsonObject task = tasksPerRoom.getAsJsonObject(taskId);
+                    String caregiverId = task.get("caregiverID").toString();
+                    String taskName = task.get("name").toString();
+                    String taskNote = task.get("notes").toString();
+                    String assignedStatus = task.get("assignedStatus").toString();
+                    String completionStatus = task.get("completionStatus").toString();
+
+                    // Only assigned task where assignedStatus is equal to true
+                    if (assignedStatus.equals("true")){
+                        Task t = new Task(roomStr, caregiverId, taskId, taskName,
+                                taskNote, assignedStatus, completionStatus);
+                        tasks.add(t);
+                    }
+                }
+            }
+        }
+        return tasks;
+    }
+
+    protected void displayTaskList(List < Task > tasks) {
+        List < Map < String,  String >> data = new ArrayList < Map < String, String >> ();
+        for (Task t: tasks) {
+            Map < String,
+                    String > room = new HashMap < String,
+                    String > (2);
+            room.put("title", t.taskName.replace("\"", " "));
+            room.put("subtitle", " " + t.room);
+            data.add(room);
+        }
+        final ListView list = view.findViewById(R.id.caregiveeTaskList);
+        SimpleAdapter adapter = new SimpleAdapter(
+                getActivity(), data, android.R.layout.simple_list_item_2, new String[] {
+                "title",
+                "subtitle"
+        },
+                new int[] {
+                        android.R.id.text1,
+                        android.R.id.text2
+                });
+        list.setAdapter(adapter);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        view = inflater.inflate(R.layout.fragment_task__caregivee, container, false);
```

Comment:
Also what does this do lol I've been seeing it everywhere haha

Reply:
It an automatic built-in function for the page. I don't think we can change that since that represent the view layout